File: 프로그래머스/unrated/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.py
def solution(arr, k):
    answer = [arr[0]]
    k -= 1
    for i in range(1, len(arr)):
        if k == 0:
            return answer
        
        if arr[i] not in answer:
            answer.append(arr[i])
            k -=1
    if k == 0:
        return answer
    else:
        return answer + [-1]*k
    

File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int k) {
        List<Integer> answer = new ArrayList<>();
        answer.add(arr[0]);
        k -= 1;
        for (int i = 1; i < arr.length; i++) {
            if (k == 0) {
                break;
            }

            if (!answer.contains(arr[i])) {
                answer.add(arr[i]);
                k -= 1;
            }
        }
        
        for (int i = 0; i < k; i++) {
            answer.add(-1);
        }

        return answer.stream().mapToInt(v -> v).toArray();
    }
}